### PR TITLE
Add geodistance query documentation

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Add geodistance query documentation

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _query-dsl/geo-and-xy/geodistance.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+## Request fields
+
+Geodistance queries accept the following fields.
+
+Field | Data type | Description
+:--- | :--- | :--- 
+`_name` | String | The name of the filter. Optional.
+`distance` | String | The distance within which to match the points. This distance is the radius of a circle centered at the specified point. For supported distance units, see [Distance units]({{site.url}}{{site.baseurl}}/api-reference/common-parameters/#distance-units). Required.
+`distance_type` | String | Specifies how to calculate the distance. Valid values are `arc` or `plane` (faster but inaccurate for long distances or points close to the poles). Optional. Default is `arc`.
```
Comment (Naarcha-AWS) [inline _query-dsl/geo-and-xy/geodistance.md:115]: What does `arc` do?

File: _query-dsl/geo-and-xy/geodistance.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+`validation_method` | String | The validation method. Valid values are `IGNORE_MALFORMED` (accept geopoints with invalid coordinates), `COERCE` (try to coerce coordinates to valid values), and `STRICT` (return an error when coordinates are invalid). Optional. Default is `STRICT`.
+`ignore_unmapped` | Boolean | Specifies whether to ignore an unmapped field. If set to `true`, the query does not return any documents that have an unmapped field. If set to `false`, an exception is thrown when the field is unmapped. Optional. Default is `false`.
```
Comment (Naarcha-AWS) [inline _query-dsl/geo-and-xy/geodistance.md:117]: Does this setting work with multiple fields?

**Review by Naarcha-AWS — APPROVED** [review]
A couple of questions.

File: _query-dsl/geo-and-xy/geodistance.md (hunk 1 above)
Comment (kolchfa-aws) [reply _query-dsl/geo-and-xy/geodistance.md:115]: Arc is distance is a way to calculate distance between two points on a sphere. https://en.wikipedia.org/wiki/Geographical_distance

File: _query-dsl/geo-and-xy/geodistance.md (hunk 2 above)
Comment (natebower) [inline _query-dsl/geo-and-xy/geodistance.md:117]:
```suggestion
`ignore_unmapped` | Boolean | Specifies whether to ignore an unmapped field. If set to `true`, then the query does not return any documents that contain an unmapped field. If set to `false`, then an exception is thrown when the field is unmapped. Optional. Default is `false`.
```

File: _query-dsl/geo-and-xy/geodistance.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+## Accepted formats
+
+You can specify the geopoint coordinates when indexing a document and searching for documents in any [format]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/geo-point#formats) that the geopoint field type accepts.
```
Comment (natebower) [inline _query-dsl/geo-and-xy/geodistance.md:121]:
```suggestion
You can specify the geopoint coordinates when indexing a document and searching for documents in any [format]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/geo-point#formats) accepted by the geopoint field type.  
```

File: _query-dsl/geo-and-xy/index.md [hunk 4]
```diff
@@ -24,13 +24,13 @@ xy queries return documents that contain:
...
 OpenSearch provides the following geographic query types:
 
 - [**Geo-bounding box queries**]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/geo-and-xy/geo-bounding-box/): Return documents with geopoint field values that are within a bounding box. 
-- **Geodistance queries** return documents with geopoints that are within a specified distance from the provided geopoint.
-- **Geopolygon queries** return documents with geopoints that are within a polygon.
-- **Geoshape queries** return documents that contain:
+- [**Geodistance queries**]({{site.url}}{{site.baseurl}}/query-dsl/geo-and-xy/geodistance/): Return documents with geopoints that are within a specified distance from the provided geopoint.
+- **Geopolygon queries**: Return documents with geopoints that are within a polygon.
+- **Geoshape queries**: Return documents that contain:
     - geoshapes and geopoints that have one of four spatial relations to the provided shape: `INTERSECTS`, `DISJOINT`, `WITHIN`, or `CONTAINS`.
```
Comment (natebower) [inline _query-dsl/geo-and-xy/index.md:35]: Apologies for commenting outside the new content, but these list items should be initially capitalized.

File: _query-dsl/geo-and-xy/xy.md [hunk 5]
```diff
@@ -12,13 +12,13 @@ redirect_from:
 
 # xy query
 
-To search for documents that contain [xy point]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/xy-point) and [xy shape]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/xy-shape) fields, use an xy query. 
+To search for documents that contain [xy point]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/xy-point/) and [xy shape]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/xy-shape/) fields, use an xy query.
```
Comment (natebower) [inline _query-dsl/geo-and-xy/xy.md:15]: "and" => "or"?

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _query-dsl/geo-and-xy/xy.md (hunk 5 above)
Comment (kolchfa-aws) [inline _query-dsl/geo-and-xy/xy.md:15]:
```suggestion
To search for documents that contain [xy point]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/xy-point/) or [xy shape]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/xy-shape/) fields, use an xy query. 
```

**Pull request merged**